File: Cursojava22/app/src/main/java/com/example/cursojava2/UsuariosFirebase/SecondActivity.java
package com.example.cursojava2.UsuariosFirebase;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.example.cursojava2.R;
import com.google.firebase.auth.FirebaseAuth;

public class SecondActivity extends AppCompatActivity {
    private FirebaseAuth firebaseAuth;
    private Button logout;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);


        firebaseAuth = FirebaseAuth.getInstance();

        logout = (Button) findViewById(R.id.btnLogout);

        logout.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Logout();
            }
        });
    }

    private void Logout() {
        firebaseAuth.signOut();
        finish();
        startActivity(new Intent(SecondActivity.this, LoginFire.class));
    }
}

File: KotlinCurso2/app/src/main/java/com/example/kotlincurso2/UsuariosFirebase/SecondActivity.kt
package com.example.kotlincurso2.UsuariosFirebase

import android.content.Intent
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import com.example.kotlincurso2.R
import com.google.firebase.auth.FirebaseAuth

class SecondActivity : AppCompatActivity() {
    private var firebaseAuth: FirebaseAuth? = null
    private var logout: Button? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_second)


        firebaseAuth = FirebaseAuth.getInstance()

        logout = findViewById(R.id.btnLogout) as Button

        logout!!.setOnClickListener(View.OnClickListener { Logout() })
    }

    private fun Logout() {
        firebaseAuth?.signOut()
        finish()
        startActivity(Intent(this@SecondActivity, LoginFire::class.java))
    }
}